lib/browser.ts

```diff
@@ -589,10 +589,16 @@ export class ProtractorBrowser extends Webdriver {
    * @returns {!webdriver.promise.Promise} A promise that will resolve to whether
    *     the element is present on the page.
    */
-  isElementPresent(locatorOrElement: ProtractorBy|WebElement): wdpromise.Promise<any> {
-    let element =
-        ((locatorOrElement as any).isPresent) ? locatorOrElement : this.element(locatorOrElement);
-    return (element as any).isPresent();
+  isElementPresent(locatorOrElement: Locator|WebElement): wdpromise.Promise<any> {
+    let element: ElementFinder;
+    if ((locatorOrElement as any).isPresent) {
```

Comment:
Either the method signature has to also include ElementFinder or we should remove this conditional.

Add in ElementFinder to signature:
`locatorOrElement: Locator|ElementFinder|WebElement`

Change conditional for ElementFinder:
`if (locatorOrElement instanceof ElementFinder) {`